CSS & JS for Bok 1 Any% exp table

xDrHellx opened the pull request:
## Files
- Added CSS file containing classes for using Boktai element colors on texts
- Added CSS & JS files related to the Any% exp table

## JavaScript
- Added a JS object used for handling the Any% exp table
- Added a function in said object for adding text-color classes on the corresponding columns of each row to make things easier to read and keep track of

xDrHellx committed:
Separated Bok 1 Any% exp table & css

- Separated the Bok 1 Any% exp table to its own file to make it easier to update & read
- Added a CSS file for the exp table
- Added CSS file inclusion

xDrHellx committed:
Added CSS for Boktai element colors

- Added a CSS file containing classes for coloring text with Boktai element colors

xDrHellx committed:
Added JS for Bok 1 Any% exp table & colors

- Added a JS file that will be used for any JS related to the Any% exp table

- Added JS functions to add classes on columns related to Lens exp, to make things easier to read and keep track of

raphaelr commented on theme/js/any-percent-exp-table.js:
```diff
@@ -0,0 +1,60 @@
+                    switch(c) {
+                        case 1:         // Sol lens EXP
+                        case 2:
+                            columns[c].classList.add("c-sol");
```
Can this be done with css and the `:nth-child` selector instead? Using .css for styling is preferable to using .js

xDrHellx replied:
I've switched to CSS to do this, see https://github.com/shenef/Boktai-Speedrunning/pull/1/commits/1f28364ff1b17e0e490083c52491f139cf9ac30c

raphaelr commented on theme/css/colors.css:
```diff
@@ -0,0 +1,28 @@
+/* Elements-related colors */
+.c-sol {
```
Do these colors look good with both the default and the dark mdbook themes?

xDrHellx replied:
Is this good enough?
I've adjusted the .c-dark color, it was too dark to be read properly on dark themes.
![elem_colors](https://github.com/shenef/Boktai-Speedrunning/assets/17726521/6e483e0b-c928-4e19-853b-2b54d238f2b0)

xDrHellx committed:
Replaced .c-dark color

xDrHellx committed:
Switched to CSS for setting texts colors

- Added CSS for setting column texts colors related to Lens exp
- Removed JS function for adding the texts-related classes, as it is not useless

shenef committed:
use css variables for colors

shenef committed:
remove leftover file

shenef committed:
adjust some comments, text and formatting
